### PR TITLE
[headnode metric graphs] add disk and memory graphs for headnode CloudWatch dashboard

### PR DESCRIPTION
**Pull request by delongmeng**
### Description of changes
* Add new graphs `Disk Used Percent` and `Memory Used Percent` to CloudWatch Dashboard of a cluster to monitor memory usage and disk usage of the root volume on the head node.

### Tests
* Manual test: 
  * After code change, create a new cluster, and verify that new graphs `Disk Used Percent` and `Memory Used Percent` (within the `Head Node EC2 Metrics` - `Head Node Instance Metrics` section) show up in CloudWatch Dashboard as expected. 
  * For a head node of c5.large Nitro instance, "reboot" of the head node didn't affect the disk usage metric and graph.
  * Update the cluster to have Monitoring - Dashboards - CloudWatch disabled using the following configuration section, then observed that the dashboard was deleted; update the cluster again with default configuration (Enabled: true), and observed that the dashboard showed up again.
```
Monitoring:
  Dashboards:
    CloudWatch:
      Enabled: false  # true
```
* Unit tests added to verify the dashboard and graphs exist in the CloudFormation template when `Monitoring` - `Dashboards` - `CloudWatch` - `Enabled` is set to `true` (default) in the cluster configuration file.

### References
* PR that adds the disk and memory metrics: https://github.com/aws/aws-parallelcluster-cookbook/pull/1732
* PR that adds the disk and memory alarms: https://github.com/aws/aws-parallelcluster/pull/4877

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by chenwany** [comment]
Could you perform a manually test to verify on Nitro instances, for example c5.large, after headnode reboot, the disk_used_percent still get value and the alarm still work?

**Review by chenwany — COMMENTED** [review]
LGTM! Only left one comment about manually test to verify disk_used_percent on Nitro instances

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,8 @@ CHANGELOG
 
 3.6.0
 ----
+**ENHANCEMENTS**
+- Add CloudWatch metrics, dashboard graphs and alarms to monitor head node memory and root volume disk utilization.
```
Comment (chenwany) [inline CHANGELOG.md]: What about mentioning these two metrics in the changelog: Add `mem_used_percent` and `disk_used_percent` metrics for head node memory and root volume disk utilization tracking on the ParallelCluster CloudWatch dashboard, and set up alarms for monitoring these metrics.

**Comment by delongmeng** [comment]
> Could you perform a manually test to verify on Nitro instances, for example c5.large, after headnode reboot, the disk_used_percent still get value and the alarm still work?

Yes I tried reboot a c5.large headnode and it didn't affect the disk metric and graph/alarm.

File: cli/src/pcluster/templates/cw_dashboard_builder.py [hunk 2]
```diff
@@ -289,21 +289,34 @@ def _add_head_node_instance_metrics_graphs(self):
...
+                title="Disk Read/Write Bytes", metrics=["DiskReadBytes", "DiskWriteBytes"], namespace="AWS/EC2"
+            ),
+            new_pcluster_metric(
+                title="Disk Read/Write Ops", metrics=["DiskReadOps", "DiskWriteOps"], namespace="AWS/EC2"
+            ),
+        ]
+
+        # CW Agent metrics for graph for head node instance
+        cwagent_metrics = [
+            new_pcluster_metric(title="Disk Used Percent", metrics=["disk_used_percent"], namespace="CWAgent"),
```
Comment (chenwany) [inline cli/src/pcluster/templates/cw_dashboard_builder.py]: note: if we decide to change the aggregation metrics for disk_used_percent, we will need to update here accordingly

**Review by chenwany — APPROVED** [review]
LGTM!

**Pull request merged**